null pointer dereference in src/openrct2-ui/windows/Ride.cpp

### Operating System

Linux

### OpenRCT2 build

0.4.32

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

This is a development issue

### Describe the issue

Compiling OpenRCT2 0.4.32 errors out with a null pointer deference error:

```
In member function 'void OpenRCT2::Ui::Windows::RideWindow::ColourOnDrawEntrancePreview(OpenRCT2::Drawing::RenderTarget&, const Ride*, const OpenRCT2::Widget&)',
    inlined from 'void OpenRCT2::Ui::Windows::RideWindow::ColourOnDrawSecondaryPreview(OpenRCT2::Drawing::RenderTarget&, const Ride*)' at /build/openrct2-0.4.32+ds/src/openrct2-ui/windows/Ride.cpp:4970:40,
    inlined from 'void OpenRCT2::Ui::Windows::RideWindow::ColourOnDraw(OpenRCT2::Drawing::RenderTarget&)' at /build/openrct2-0.4.32+ds/src/openrct2-ui/windows/Ride.cpp:4842:41:
/build/openrct2-0.4.32+ds/src/openrct2-ui/windows/Ride.cpp:4976:54: error: null pointer dereference [-Werror=null-dereference]
 4976 |             if (stationObj == nullptr && stationObj->entranceBackIndex == kImageIndexUndefined)
      |                                          ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

I suspect `stationObj == nullptr` was meant to be `stationObj != nullptr`.

### Steps to reproduce

Compile the current release.

### Attachments

_No response_

--- a/src/openrct2-ui/windows/Ride.cpp
+++ b/src/openrct2-ui/windows/Ride.cpp
@@ -4987,7 +4987,7 @@ namespace OpenRCT2::Ui::Windows
         void ColourOnDrawEntrancePreview(RenderTarget& rt, const Ride* ride, const Widget& widget)
         {
             auto stationObj = ride->getStationObject();
-            if (stationObj == nullptr && stationObj->entranceBackIndex == kImageIndexUndefined)
+            if (stationObj == nullptr || stationObj->entranceBackIndex == kImageIndexUndefined)
                 return;
 
             auto trackColour = ride->trackColours[0];

